src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerDiagnosticListener.cs

```diff
@@ -372,4 +381,39 @@ static bool TryFetchException(object payload, out Exception exc)
             return true;
         }
     }
+
+    private static string GetErrorType(Exception exc)
+    {
+#if NET8_0_OR_GREATER
+        // For net8.0 and above exception type can be found using HttpRequestError.
+        // https://learn.microsoft.com/dotnet/api/system.net.http.httprequesterror?view=net-8.0
+        var httpRequestException = exc as HttpRequestException;
+        if (httpRequestException != null)
```

Comment:
nit: This can be folded into a single line with the pattern match syntax:

`if (exc is HttpRequestException httpRequestException)`